recovery_backend/backend/configmap.go

```diff
@@ -69,110 +71,120 @@ func (rb *RecoveryBackend) CreateConfigMap(hostname, version string) error {
 		}
 	}
 
-	return nil
+	_, err = datastore.VerifyCreation(configMapName, "configmap", func(name string) (runtime.Object, error) {
+		return rb.Datastore.GetConfigMapRO(rb.Namespace, name)
+	})
+	return err
 }
 
 func (rb *RecoveryBackend) EndGrace(hostname, version string) error {
 	configMapName := types.GetConfigMapNameFromHostname(hostname)
-	cm, err := rb.Datastore.GetConfigMap(rb.Namespace, configMapName)
-	if err != nil {
-		return errors.Wrapf(err, "failed to get the configmap %v", configMapName)
-	}
 
-	oldVersion := cm.Annotations["version"]
+	_, err := util.RetryOnConflictCause(func() (interface{}, error) {
+		cm, err := rb.Datastore.GetConfigMapWithoutCache(rb.Namespace, configMapName)
```

Comment:
Why we need retry here? 

If this operation fails somehow, should the recovery backend retry to call this operation again and it should succeed eventually?

Just noticed we do the same for longhorn-manager REST APIs, but why?

Reply:
cc @joshimoo @shuo-wu

Reply:
> Why we need retry here?

Because we have multiple recovery-backend pod, not sure if all clients update the configmap from the smae pod.

Reply:
> If this operation fails somehow, should the recovery backend retry to call this operation again and it should succeed eventually?

3 replaces can do retry
1. NFS-ganesha: it doesn't do retry from the observation. This will make the connection failed.
2. api/route: the logic is incorrect https://github.com/longhorn/longhorn/issues/5259
3. Current implementation

3. makes sense for the update conflict error.

Reply:
Due to the buggy router retry mechanism, we should retain this kind of retry. After fixing https://github.com/longhorn/longhorn/issues/5259, we can consider removing it.